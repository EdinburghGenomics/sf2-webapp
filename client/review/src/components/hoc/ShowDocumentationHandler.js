// @flow
import * as React from 'react';

import DocumentationModal from "../general/DocumentationModal";


type SF2Props = { showDocumentation: () => void }


export const SF2DefaultProps = {
    showDocumentation: () => {}
};


export const withShowDocumentationHandler = (WrappedSF2 : React.ComponentType<SF2Props>) : React.ComponentType<SF2Props> => {


    type ShowDocumentationHandlerState = {
        documentationModalActive: boolean
    };


    class ShowDocumentationHandler extends React.Component<SF2Props, ShowDocumentationHandlerState> {
        static defaultProps = SF2DefaultProps;

        constructor(props) {
            super(props);
            this.state = {
                documentationModalActive: false
            };
        }


        showDocumentation = () : void => {
            this.setState({
                documentationModalActive: true
            });
        };


        hideModals = () => {
            this.setState({
                documentationModalActive: false
            });
        };


        onCancel = () => {
            this.setState({
                documentationModalActive: false
            });
        };


        render() {

            const {
                showDocumentation,
                ...passThroughProps
            } = this.props;

            return (
                <div>
                <DocumentationModal onCancel={this.hideModals} active={this.state.documentationModalActive}/>
                <WrappedSF2
                    showDocumentation={this.showDocumentation}
                    {...passThroughProps}
                />
                </div>
            );

        }
    }

    return ShowDocumentationHandler;

};
