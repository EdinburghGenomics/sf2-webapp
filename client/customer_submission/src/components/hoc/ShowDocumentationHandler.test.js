import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { withShowDocumentationHandler } from './ShowDocumentationHandler';


it('renders the DocumentationViewer correctly with required arguments', () => {

    const wrappedComponent = props => <div>test</div>;
    const fullComponentClass = withShowDocumentationHandler(wrappedComponent);
    const fullComponentInstance = new fullComponentClass({});

    const wrapper = mount(fullComponentInstance.render());

    expect(toJson(wrapper)).toMatchSnapshot();

});
