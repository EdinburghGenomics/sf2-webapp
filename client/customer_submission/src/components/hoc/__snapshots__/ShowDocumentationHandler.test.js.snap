// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DocumentationViewer correctly with required arguments 1`] = `
<div>
  <DocumentationModal
    active={false}
    onCancel={[Function]}
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      style={
        Object {
          "minWidth": "95%",
        }
      }
      zIndex={1050}
    />
  </DocumentationModal>
  <wrappedComponent
    showDocumentation={[Function]}
  >
    <div>
      test
    </div>
  </wrappedComponent>
</div>
`;
